src/auth/connection.ts

```diff
@@ -23,10 +23,37 @@ export const ssoAccountAccessScopes = ['sso:account:access']
 export const codewhispererScopes = ['codewhisperer:completions', 'codewhisperer:analysis']
 export const defaultSsoRegion = 'us-east-1'
 
+type SsoType =
+    | 'any' // any type of sso
+    | 'base' // AWS Identity Center
+    | 'builderId'
+
 export const isIamConnection = (conn?: Connection): conn is IamConnection => conn?.type === 'iam'
-export const isSsoConnection = (conn?: Connection): conn is SsoConnection => conn?.type === 'sso'
-export const isBuilderIdConnection = (conn?: Connection): conn is SsoConnection =>
-    isSsoConnection(conn) && conn.startUrl === builderIdStartUrl
+export const isSsoConnection = (conn?: Connection, type: SsoType = 'any'): conn is SsoConnection => {
+    if (conn?.type !== 'sso') {
+        return false
+    }
+    // At this point the conn is an SSO conn, but now we must determine the specific type
+    switch (type) {
+        case 'base':
+            // Any SSO type is considered an SSO connection, but in this case
+            // we only want an Identity Center connection. So we return true
+            // as long as the connection is none of the other connection types.
+
+            // IMPORTANT: This condition should grow as more SsoType's get added.
+            return !isBuilderIdConnection(conn)
+        case 'builderId':
+            return conn.startUrl === builderIdStartUrl
+        case 'any':
+            return true
+    }
+}
+export const isAnySsoConnection = (conn?: Connection): conn is SsoConnection => isSsoConnection(conn, 'any')
+export const isBaseSsoConnection = (conn?: Connection): conn is SsoConnection => isSsoConnection(conn, 'base')
```

Comment:
- Ideally we would avoid "enterprise sso" or "idc" or other marketing names, but using yet another term "base" probably is a higher cost than just using "idc".
- If wonder if we'll later need IdC sub-types like `idc.federated`.


```suggestion
export const isIdcSsoConnection = (conn?: Connection): conn is SsoConnection => isSsoConnection(conn, 'idc')
```

Reply:
Updated